lib/addons/p5.dom.js

```diff
@@ -2880,7 +2880,8 @@
     this.setModified(true);
     return this;
   };
-  p5.MediaElement.prototype.get = function(x, y, w, h) {
+  p5.MediaElement.prototype.get = p5.Renderer2D.prototype.get;
```

Comment:
Why not return in the same fashion as p5.Image.get for consistency?

Reply:
i think it's clearer this way to see that it's just reusing another classes implementation.

i think this is preferable to the way that p5.Image does it since in order to determine what it's doing, you have to first parse the code, work out if the argument lists are the same, determine whether or not it matters that the argument list is changed, etc...

Reply:
Thanks for the explanation! I know it is breaking from the repo's convention a bit but I agree that this is clearer and we should probably move towards doing this wherever it's applicable. My only concern would be that the single-line approach disappears a little bit in the organization of the code-base when visually scanning, maybe just something that would eventually require consistent placement if this is done a lot. I say change `p5.Image.prototype.get` to match this style but maybe we should get a second opinion first. @outofambit what do you think?

Reply:
I think a better solution here would to have the implementation of `p5.Renderer.prototype.get` be unattached to any class implementation (and so just be `get` in its own file). Each class that uses it could pass in all the necessary data. it looks like that is only:

* `x, y, w, h`
* `pixelsState`
* `canvas`
* `getPixel`?

or maybe

* `x, y, w, h`
* `this`

Each class that uses this implementation (p5.Renderer, p5.Image, p5.MediaElement) could internally reference this single implementation. It looks like a similar pattern would work for `getPixel` as well.

What do you think @Spongman and @stalgiag? Happy to explain more if I'm not totally making sense.

Reply:
i would rather keep the implementation in a common base class. the fact that `Image` & `MediaElement` have to 'borrow' the `get` implementation from outside their hierarchy is something i'd rather solve by cleaning up their hierarchies. IMO it's better to move toward increased encapsulation rather than away.

Reply:
I think I agree with @Spongman here. I can imagine ways in which what you propose would be nice, with shared functions being detached from any class, but my personal feeling is that this would be moving towards a pretty big change for how the library is structured which is something I am always reluctant about.

Reply:
Fair enough. If this pattern is present outside of this PR I'm ok with treading water here. Just to be clear though, my suggested implementation is a move towards encapsulation and away from implicit assumptions about classes and objects in the implementation of `get` and `_getPixel`.